core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -121,6 +124,36 @@
   private SqlFunctions() {
   }
 
+  /** SQL TO_BASE64(string) function*/
+  public static String toBase64(String string) {
+    if (string == null) {
+      return null;
+    }
+    String base64 = new String(Base64.getEncoder().encode(string.getBytes(UTF_8)), UTF_8);
```

Comment:
you can use [Encoder.encodeToString()](https://docs.oracle.com/javase/8/docs/api/java/util/Base64.Encoder.html#encodeToString-byte:A-) function